imap/jmap_ical.c

```diff
@@ -241,13 +241,22 @@ static const char *sha1key(const char *val, char *keybuf)
 
 static char *normalized_uri(const char *uri)
 {
-    const char *col = strchr(uri, ':');
-    if (!col) return xstrdupnull(uri);
+    if (!uri) return NULL;
 
     struct buf buf = BUF_INITIALIZER;
-    buf_setmap(&buf, uri, col - uri);
-    buf_lcase(&buf);
-    buf_appendcstr(&buf, col);
+    const char *col = strchr(uri, ':');
```

Comment:
this could probably use a comment along the lines of "lowercase just the uri scheme" to explain why we're splitting at the colon.

Reply:
Right, added a comment

Reply:
And the next for code style review will be to decide if a comment should go above or inside the `if` block :)